src/main/java/seedu/duke/storage/Store.java

```diff
@@ -67,20 +83,42 @@ private void createFileIfNotExist (String fullPath) throws IOException {
      *
      * @param accounts The AccountList instance
      */
-    public void loadAccountsFromStore (AccountList accounts) throws Exception {
+    public void loadAccountsFromStore (AccountList accounts) throws IOException {
         String fullPath = directory + ACCOUNTS_FILE_NAME;
         createFileIfNotExist(fullPath);
         BufferedReader br = new BufferedReader(new FileReader(fullPath));
-        Storage[] store = gson.fromJson(br, Storage[].class);
+        Storage[] store = new Storage[]{};
+        try {
+            store = gson.fromJson(br, Storage[].class);
+        } catch (Exception e) {
+            ui.printMessage(ErrorMessage.ERR_LOADING_ACCOUNTS);
+            createFile(fullPath);
+        }
+
         for (Storage account : store) {
             Currency currency = Currency.valueOf(account.getCurrency());
             float value = account.getValue() / 100.0f;
             logger.log(Level.INFO, "Loaded account " + currency + " with value " + value);
-            accounts.addAccount(currency, value);
+            try {
+                if (value < 0) {
+                    ui.printMessage(ErrorMessage.ACCOUNT_BALANCE_NEGATIVE_WHEN_LOADING);
+                    value = 0;
+                }
+                if (value > Account.UPPER_BOUND) {
+                    ui.printMessage(ErrorMessage.ACCOUNT_BALANCE_TOO_LARGE_WHEN_LOADING);
+                    value = (float) Account.UPPER_BOUND;
+                }
+                accounts.addAccount(currency, value);
+            } catch (AccountAlreadyExistsException e) {
+                logger.log(Level.WARNING,
+                        "Skipping duplicate account " + currency + " with value " + value + " found in store");
+                ui.printf(ErrorMessage.DUPLICATE_ACCOUNT_WHEN_LOADING, currency);
+            }
```

Comment:
All of these checks exist within the Validator class. Might it be possible to run the parsed account value through the validator class? You also do not account for checking for non-numeric value, which could be an issue. Either way, great implementation as is, use the validator if you feel it is necessary. Great job!

Reply:
Thank you for bringing up the point on non-numeric values! After some consideration, I have decided not to use the Validator here for 2 reasons. 
1. I need to implement custom logic for both potential errors, either setting the balance to 10 million if it is too large or 0 if it is negative. The validator throws the same exception with different error messages, and working around it would probably lead to some janky logic. 
2. The gson parser already handles the potential non-numeric cases by throwing a parse error. We currently handle that by wiping the file as it is deemed an "irrecoverable" error. We also avoid precision errors as we store the balances as integers instead of floats, so it cannot be edited to have more than 3 decimal places.